src/branchandbound.jl

```diff
@@ -1,101 +1,59 @@
 # univariate
-function enclose_BranchandBound(f::Function, dom::Interval; order=10, tol=0.6)
-    x0 = Interval(mid(dom))
-    x = TaylorModel1(order, x0, dom)
-    return branchandbound(f(x-x0).pol, dom, tol)
-end
 
-# multivariate
-function enclose_BranchandBound(f::Function, dom::IntervalBox{D};
-                                order=10, tol=0.6) where {D}
-    x0 = mid(dom)
-    set_variables(Float64, "x", order=2order, numvars=D)
-    x = [TaylorModelN(i, order, IntervalBox(x0), dom) for i=1:D]
-    return branchandbound(f(x...).pol, dom - x0, tol)
+@inline function enclose(f::Function, X::Interval, ba::BranchAndBoundEnclosure;
```

Comment:
I think for `enclose` it's ok, since all it does is calling the `_enclose` function, so you don't get any code duplication overhead. (see [here](https://en.wikipedia.org/wiki/Inline_expansion#Effect_on_performance) for a nice overview of pros and cons of inlining), but I'm not expert either, any reason why you thought this shouldn't be inlined?

Reply:
for `monotonicity_check` I am not that sure, might be better *not* to inline that, since `_branch_and_bound` is recursive and that's called at each iteration

Reply:
sounds good